cmd/thv-operator/controllers/mcpregistry_controller.go

```diff
@@ -192,11 +352,32 @@ func (r *MCPRegistryReconciler) reconcileSync(ctx context.Context, mcpRegistry *
 	return result, err
 }
 
+// isAPIReady checks if the registry API deployment is ready and serving requests
+func (r *MCPRegistryReconciler) isAPIReady(ctx context.Context, mcpRegistry *mcpv1alpha1.MCPRegistry) bool {
+	ctxLogger := log.FromContext(ctx).WithValues("mcpregistry", mcpRegistry.Name)
+
+	deploymentName := fmt.Sprintf("%s-api", mcpRegistry.Name)
```

Comment:
maybe this and similar name generators will go to dedicated functions

Reply:
done locally